packages/slice-machine/components/ScreenshotChangesModal/DropZone.tsx

```diff
@@ -0,0 +1,88 @@
+import { ReactNode, useState } from "react";
+import { Flex } from "theme-ui";
+import { acceptedImagesTypes } from "@lib/consts";
+
+const sharedFlexSx = {
+  justifyContent: "center",
+  alignItems: "center",
+  height: "100%",
+  width: "100%",
+};
+
+interface DropZoneProps {
+  isVisible?: boolean;
+  onHandleDrop: (file: File) => void;
+  imageTypes?: string[];
+  children: ReactNode | ((options: { isDragActive: boolean }) => ReactNode);
+}
+
+const DropZone: React.FC<DropZoneProps> = ({
+  children,
+  onHandleDrop,
+  isVisible = false,
+  imageTypes = acceptedImagesTypes,
+}) => {
+  const [isDragActive, setIsDragActive] = useState(false);
+
+  const handleDrop = (e: React.DragEvent<HTMLInputElement>) => {
```

Comment:
OPT: for readability I think it's worth giving `e` a more explicit name, like `event` or `dragEvent`, especially when it's used over a few different lines